flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/FlinkTypeFactory.scala

```diff
@@ -123,6 +123,9 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem) extends JavaTypeFactoryImp
           true,
           getTypeSystem)
 
+      case LogicalTypeRoot.SYMBOL =>
+        createSqlType(SqlTypeName.SYMBOL)
```

Comment:
This means we do not preserve the kind of symbol? Maybe you can also add a comment or TODO here, because information is also lost here.

Reply:
Calcite's `SYMBOL` type does not store the class/kind of the symbol in the type itself. It is actually a pseudo type if I understand it correctly. This type can be used only for literals and therefore the class/kind is already part of the stored value. 

So there is no reason to store it separately imo. Or do you think we need our own `SymbolRelDataType`?

Reply:
Either we need our own `SymbolRelDataType` or we need to adapt our logical type. This can be a follow up. I just would like to add a TODO or JIRA issue to not forget about it.

Reply:
Sure I can add a follow-up. I think though we should rather adapt our `SymbolType`.

Reply:
I created https://issues.apache.org/jira/browse/FLINK-17151